### PR TITLE
Editor: Prevents the editor from hanging on Share text and Reblog actions

### LINKED ISSUE #14972: Reader: re-blogging causes editor to display loading indicator indefinitely
### Expected behavior
When tapping the re-blog button in the Reader, a post is created and I see the re-blogged content in the editor. 

### Actual behavior
When tapping the re-blog button in the Reader, a post is created, but displays a loading indicator indefinitely. Tapping the back button and navigating to post Drafts does successfully open the post editor with the expected content. 

### Steps to reproduce the behavior
1. Launch WPAndroid. 
2. Navigate to Reader. 
3. Tap the re-blog button. 

##### Tested on Samsung Galaxy S20, Android 11, WPAndroid 17.7-rc-1. 

The video below showcases the unexpected experience. 

https://user-images.githubusercontent.com/438664/124134576-965b5300-da48-11eb-9851-0cb98a5bca24.mp4

### PR DESCRIPTION
**Pull request by antonis**
Fixes #14972 and https://github.com/wordpress-mobile/gutenberg-mobile/issues/4040

### Description

Moves shared text and reblog content setup on post creation to prevent editor from hanging

#### Share action
1. Select some text (e.g. on the browser)
2. Choose the share action on the context menu and select share
3. Select the WordPress app from this PR
4. Select a site
5. *Verify* that the editor opens with the correct shared content

#### Reblog action
1. Launch WPAndroid
2. Navigate to Reader
3. Find a post and tap the re-blog button
4. Select a site and press the check button at the top-right
5. *Verify* that the editor opens with the reblogged post content

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Relied on existing tests [`ReblogUseCaseTest`](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/test/java/org/wordpress/android/ui/reader/reblog/ReblogUseCaseTest.kt) and [`ReblogUtilsTest`](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/test/java/org/wordpress/android/util/ReblogUtilsTest.kt)

3. What automated tests I added (or what prevented me from doing so)
The code changes on the `EditPostActivity` are not easily testable

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: RELEASE-NOTES.txt
Comment (antonis) [inline RELEASE-NOTES.txt:5]: - [x] @antonis to bump the milestone before merging

**Review by develric — APPROVED** [review]
This is working as described @antonis , thanks for taking care of it 🙇 ! Also code LGTM; feel free to merge once the release note is updated based on target 👍

**Comment by antonis** [comment]
Thank you for the prompt review @develric 🙇‍♂️ 

> feel free to merge once the release note is updated based on target 👍

I'll update the release notes and merge 👍

**Pull request merged**
